src/MagnumPlugins/BasisImporter/BasisImporter.h

```diff
@@ -388,6 +394,30 @@ class MAGNUM_BASISIMPORTER_EXPORT BasisImporter: public AbstractImporter {
              * @ref CompressedPixelFormat::EacRG11Unorm.
              */
             EacRG = 21,
+
+            /**
+             * BC6 RGB unsigned HDR. Loaded as
+             * @ref CompressedPixelFormat::Bc6hRGBUfloat.
+             */
+            Bc6hRGB = 22,
+
+            /**
+             * ASTC 4x4 RGBA HDR. The alpha channel is always set to opaque.
+             * Loaded as @ref CompressedPixelFormat::Astc4x4RGBAF.
+             */
+            Astc4x4RGBAF = 23,
+
+            /**
+             * Uncompressed half float RGB HDR. Loaded as
+             * @ref PixelFormat::RGB16F.
+             */
+            RGB16F = 24,
+
+            /**
+             * Uncompressed half float RGBA HDR. The alpha channel is always
+             * set to opaque. Loaded as @ref PixelFormat::RGBA16F.
+             */
+            RGBA16F = 25,
```

Comment:
I'd like to have at least the RGB16F / RGBA16F target tested (as usual, so a file generated by BasisImageConverter from the EXR and compared, and then a resulting KTX2/Basis imported with BasisImporter and compared). The OpenExrImporter test files contain a HDR file, it's _really tiny_ so the block compression will likely do a total mess out of it, but better than nothing. OpenExrImporter has a [forceChannelType=FLOAT](https://doc.magnum.graphics/magnum/classMagnum_1_1Trade_1_1OpenExrImporter.html#Trade-OpenExrImporter-configuration) option to import half-floats as floats that you can use here. There's also a cubemap EXR file which is slightly bigger and could maybe get less distorted.

Or alternatively if the Basis repo contains some small enough sample HDR file with known output, use it. But I doubt that.

Besides that, BasisImporter test has `FormatData` which you could expand with the HDR formats. It doesn't verify the actual contents but could be enough to verify it doesn't enter some infinite allocation loop or whatever in these.

Reply:
I tried to cover all affected code paths, without duplicating every single test:

Importer:
- test error when transcoding LDR `<->` HDR
- test unconfigured plugin defaulting to `RGBA16F` for HDR .basis files
- transcode .basis to `RGB16F` + compare against source RGB .exr
- transcode .basis to `RGBA16F` + compare against source RGBA .exr
- transcode .basis to BC6 and ASTC4x4RGBAF

Converter
- test mip generation with HDR input
- test patching away .basis sRGB flag for HDR input - HDR can't be sRGB
- convert a 2D HDR mip chain + compare against source .exr files
- last test also tests Basis behavior of dropping the alpha channel and making it 1 always